src/guide/instance.md

```diff
@@ -31,109 +57,75 @@ Root Instance
+## Component Instance Properties
 
-```js
-// Our data object
-const data = { a: 1 }
+Earlier in the guide we met `data` properties:
 
-// The object is added to the root instance
-const vm = Vue.createApp({
+```js
+const app = Vue.createApp({
   data() {
-    return data
+    return { count: 4 }
   }
-}).mount('#app')
-
-// Getting the property on the instance
-// returns the one from the original data
-vm.a === data.a // => true
+})
 
-// Setting the property on the instance
-// also affects the original data
-vm.a = 2
-data.a // => 2
-```
+const vm = app.mount('#app')
 
-When this data changes, the view will re-render. It should be noted that properties in `data` are only **reactive** if they existed when the instance was created. That means if you add a new property, like:
+console.log(vm.count) // => 4
 
-```js
-vm.b = 'hi'
+// Assign a new value
+vm.count = 5
 ```
 
-Then changes to `b` will not trigger any view updates. If you know you'll need a property later, but it starts out empty or non-existent, you'll need to set some initial value. For example:
+As shown above, properties defined in `data` are exposed via the component instance, `vm`. Vue's reactivity system will track when those properties are used and updates will be triggered if a new value is assigned.
 
-```js
-data() {
-  return {
-    newTodoText: '',
-    visitCount: 0,
-    hideCompletedTodos: false,
-    todos: [],
-    error: null
-  }
-}
-```
-
-The only exception to this being the use of `Object.freeze()`, which prevents existing properties from being changed, which also means the reactivity system can't _track_ changes.
+To define methods on a component instance we would usually use the `methods` option:
 
 ```js
-const obj = {
-  foo: 'bar'
-}
-
-Object.freeze(obj)
-
-const vm = Vue.createApp({
+const app = Vue.createApp({
   data() {
-    return obj
+    return { count: 4 }
+  },
+  methods: {
+    increment() {
+      // `this` refers to the same object as vm
+      this.count++
+    }
   }
-}).mount('#app')
-```
+})
 
-```html
-<div id="app">
-  <p>{{ foo }}</p>
-  <!-- this will no longer update `foo`! -->
-  <button v-on:click="foo = 'baz'">Change it</button>
-</div>
-```
+const vm = app.mount('#app')
 
-In addition to data properties, instances expose a number of useful instance properties and methods. These are prefixed with `$` to differentiate them from user-defined properties. For example:
+console.log(vm.count) // => 4
 
-```js
-const vm = Vue.createApp({
-  data() {
-    return {
-      a: 1
-    }
-  }
-}).mount('#example')
+vm.increment()
 
-vm.$data.a // => 1
+console.log(vm.count) // => 5
 ```
 
-In the future, you can consult the [API reference](../api/instance-properties.html) for a full list of instance properties and methods.
+Inside the methods of a component, the `this` value will be bound to the component instance.
```

Comment:
**opinion**: this feels like we're repeating `Data and methods` part. Should we just make a list of possible component instance properties here without diving in explanation, or should we exclude this section altogether?

Reply:
Good point.

I'll need to mull this over. I don't think I'd want to remove it altogether but it might be possible to cut it back significantly to the bits that are most relevant to the topic at hand.

I need to check the surrounding pages to ensure the overall narrative still holds together, especially if we're going to move `data-methods.md` to be after `template-syntax.md`.

Reply:
I've cut a significant portion from that section. I think it's much better.